src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs

```diff
@@ -83,11 +86,13 @@ public ForkchoiceUpdatedHandler(
         _syncPeerPool = syncPeerPool;
         _simulateBlockProduction = simulateBlockProduction;
         _secondsPerSlot = secondsPerSlot;
+        _historyPruner = historyPruner;
```

Comment:
The handler is merge specific. Can you make it work without modifying merge plugin? Think about pow network or network that does not use the rpc handler for consensus.